python/ray/dag/compiled_dag_node.py

```diff
@@ -543,11 +545,38 @@ def __init__(
         # create a reader ref for a larger object when the channel backing store is
         # resized, etc.). The `_driver_actor` serves as a way for the output writer to
         # invoke remote functions on the driver node.
-        self._driver_actor = CompiledDAG.DAGDriverProxyActor.options(
-            scheduling_strategy=NodeAffinitySchedulingStrategy(
-                ray.get_runtime_context().get_node_id(), soft=False
-            )
-        ).remote()
+
+        def _get_actor_handle():
+            """
+            Get the actor handle of the current process.
+
+            Returns:
+              Return the actor handle if the current task is an actor method.
+              Return None if the current process is the driver process.
+
+            Raises:
+                NotImplementedError: If the current process is a Ray task.
+            """
+            runtime_context = ray.get_runtime_context()
+            if runtime_context.worker.mode == ray.WORKER_MODE:
+                try:
+                    return ray.get_runtime_context().current_actor
+                except RuntimeError:
+                    raise NotImplementedError(
+                        "Compiled DAGs currently require the InputNode() to be the "
+                        "driver process or an actor method. Ray task is not supported."
+                    )
+            return None
+
+        self._actor_handle = _get_actor_handle()
+        self._driver_actor = None
+
+        if self._actor_handle is None:
+            self._driver_actor = CompiledDAG.DAGDriverProxyActor.options(
```

Comment:
Why do we always need this if driver creates the DAG?

Reply:
This is the existing behavior.